app/Traits/Eloquent/Remover.php

```
@@ -132,30 +98,368 @@ public function removeAllTheEntriesMediaFoldersLocal($projectRef): void
+            unset($result, $objects);
+
+            // Optional: Force garbage collection for very large operations
+            if ($totalDeleted % 10000 === 0) {
+                gc_collect_cycles();
+            }
+
+        } while ($continuationToken);
+
+        Log::info("Total objects deleted with prefix $prefix: $totalDeleted");
+        return true;
+    }
```

PHPMD (2.15.0)

The method bulkDeleteByPrefix() has a Cyclomatic Complexity of 16. The configured cyclomatic complexity threshold is 10. (Code Size Rules)

(CyclomaticComplexity)